lib/common/hash.go

```diff
@@ -7,12 +12,26 @@ import (
 	"github.com/ethereum/go-ethereum/rlp"
 )
 
+// 32 bytes / 256 bits hash type
+type Hash = ethcommon.Hash
+
+// Set the content of the hash to be the provided binary data
+var BytesToHash = ethcommon.BytesToHash
+
+// Generate a double SHA-256 hash out of the supplied binary data
 func MakeHash(b []byte) []byte {
 	first := sha256.Sum256(b)
 	second := sha256.Sum256(first[:])
 	return second[:]
 }
 
+// Interface recognized by RLP encoder
+type Encoder = rlp.Encoder
```

Comment:
`Encoder` is not used, is it for future usage?

Reply:
Yes!